docs_headless/src/content/docs/ArrayInputBase.md

```diff
@@ -0,0 +1,177 @@
+
+If you need to disable the input, make sure the children are either `disabled` and `readOnly`:
+
+```jsx
+import { ArrayInputBase, EditBase, Form } from 'ra-core';
+import { MyFormIterator } from './MyFormIterator';
+import { DateInput } from './DateInput';
+import { NumberInput } from './NumberInput';
+import { TextInput } from './TextInput';
+
+const OrderEdit = () => (
+    <EditBase>
+        <Form>
+            <TextInput source="customer" />
+            <DateInput source="date" />
+            <div>
+                <div>Items:</div>
+                <ArrayInputBase source="items">
+                    <MyFormIterator inline disabled>
+                        <TextInput source="name" readOnly/>
+                        <NumberInput source="price" readOnly />
+                        <NumberInput source="quantity" readOnly />
+                    </MyFormIterator>
+                </ArrayInputBase>
+            </div>
+            <button type="submit">Save</button>
+        </Form>
+    </EditBase>
+);
+```
+
+## Changing An Item's Value Programmatically
+
+You can leverage `react-hook-form`'s [`setValue`](https://react-hook-form.com/docs/useform/setvalue) method to change an item's value programmatically.
+
+However you need to know the `name` under which the input was registered in the form, and this name is dynamically generated depending on the index of the item in the array.
+
+To get the name of the input for a given index, you can leverage the `SourceContext` created by react-admin, which can be accessed using the `useSourceContext` hook.
+
+This context provides a `getSource` function that returns the effective `source` for an input in the current context, which you can use as input name for `setValue`.
+
+Here is an example where we leverage `getSource` and `setValue` to change the role of an user to 'admin' when the 'Make Admin' button is clicked:
+
+```tsx
+import { ArrayInputBase, useSourceContext } from 'ra-core';
+import { useFormContext } from 'react-hook-form';
+import { MyFormIterator } from './MyFormIterator';
+
+const MakeAdminButton = () => {
+    const sourceContext = useSourceContext();
+    const { setValue } = useFormContext();
+
+    const onClick = () => {
+        // sourceContext.getSource('role') will for instance return
+        // 'users.0.role'
+        setValue(sourceContext.getSource('role'), 'admin');
+    };
+
+    return (
+        <button onClick={onClick}>
+            Make admin
+        </button>
+    );
+};
+
+const UserArray = () => (
+    <ArrayInputBase source="users">
+        <MyFormIterator inline>
+            <TextInput source="name" helperText={false} />
+            <TextInput source="role" helperText={false} />
+            <MakeAdminButton />
+        </MyFormIterator>
+    </ArrayInputBase>
+);
+```
+
+**Tip:** If you only need the item's index, you can leverage the [`useSimpleFormIteratorItem` hook](./SimpleFormIterator.md#getting-the-element-index) instead.
```

Comment:
```suggestion
**Tip:** If you only need the item's index, you can leverage the `useSimpleFormIteratorItem` hook instead.
```

(Page doesn't exist yet in ra-core, other alternative would be to link to the RA doc)